test/viewModels/routeDetails/locationOfGoods/LocationOfGoodsViewModelSpec.scala

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package viewModels.routeDetails.locationOfGoods
+
+import base.SpecBase
+import generators.{Generators, RouteDetailsUserAnswersGenerator}
+import models.reference.{CustomsOffice, UnLocode}
+import models.{Address, Coordinates, LocationOfGoodsIdentification, LocationType, PostalCodeAddress}
+import org.scalacheck.Arbitrary.arbitrary
+import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks
+import pages.routeDetails.locationOfGoods._
+
+class LocationOfGoodsViewModelSpec extends SpecBase with ScalaCheckPropertyChecks with Generators with RouteDetailsUserAnswersGenerator {
+  "apply" - {
+    "when 'v' customs office" - {
+      "must return 3 rows" in {
+        val userAnswers = emptyUserAnswers
```

Comment:
Thanks for adding this. I wonder if we could do something cool here to utilise the arbitrary answers generator. @kenneth-rayner  do you mind if I pull the branch down and try something?